### ISSUE TITLE
Story: Add External Link Icon to NCIDS Footer

### ISSUE THREAD

**Issue by andyvanavery31** [NCIOCPL/ncids#1901]
### Specifications

**Visual**
- External Link icon is center aligned vertically on the Footer links
- External Link icon is always fixed on the right
- If the link text would wrap, the last word of the title will wrap with the icon so the icon does not sit alone (if possible)

**Functional**
- If the link text does not have .gov it will get an external link icon
- A Footer link can be overridden to have an external link icon
- A Footer link with an external link icon can be overridden to remove it

**Impacted Components**

- [ ] Footer
  - [ ] [Doc Site](https://designsystem-dev.cancer.gov/develop/components/footer)
  - [ ] [Storybook Footer NCI Big](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-footer-variants-nci-big--english)

### DoR Tasks

- [x] Comps Created
- [x] Comps Reviewed using [Checklist](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Product-Review-of-Design-Solutions)
  - [x] Developer Review
  - [x] IA Review
  - [x] Product Review
- [x] Comps Approved
  - [x] Developer Approval
  - [x] IA Approval
  - [x] Product Approval
- [x] Scenarios Created
- [x] Scenarios Reviewed 
  - [x] Designer Review
  - [x] Developer Review
  - [x] IA Review
- [x] Scenarios Approved
  - [x] Designer Approval
  - [x] Developer Approval
  - [x] IA Approval
- [x] Acceptance Criteria Drafted
- [x] Acceptance Criteria Reviewed
  - [x] Developer Review
  - [x] IA Review
- [x] Acceptance Criteria Approved
  - [x] Developer Approval
  - [x] IA Approval

### DoD Tasks

- [ ] Governance Drafted
- [ ] PR Review (see PR for review and comments)
- [ ] Client review with demo

### Technical Notes

**Scenarios/Comps**

- Use the comps here: [Footer with External Link Icon](https://www.figma.com/design/GCOniv8OTyj9OC1udKwo6S/Mini-Landing-Pages?node-id=8420-41366&node-type=instance&m=dev). **NOTE**: The space between the last word of the link and the external link icon will use the same spacing as usa-link uses with the external link icon.
- On the [NCIDS Documentation Site Footer page](https://designsystem-dev.cancer.gov/develop/components/footer) add:
  - In the [Best Practices section](https://designsystem-dev.cancer.gov/develop/components/footer#best-practices), add a subsection which will be the last subsection with the heading, "External Links", and in the subsequent paragraph explain how and when to use and add the External Link Icon. In this section, explain how the external links are mostly "auto-magical", and provide an explanation of how to use the override.
  - Another [Code Snippet example](https://designsystem-dev.cancer.gov/develop/components/footer#code-snippets):
    - In the link at the same location as the one shown in the comp has an external link that links to [www.google.com](http://www.google.com/), and has the button text "Link to Google".
    - In the link after the previous one, add an external link that links to www.google.com, has the override class to not show the external link icon, and has the text, "External Link Override to Not Display"
    - In the link after the previous one, add an internal link that links to www.cancer.gov, has the override class to show the external link icon, and has the text, "Override to Display External Link Icon"
- On Storybook:
  - Within the Footer folder, add a story to the [Test Cases folder](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-footer-test-cases--no-form) called, "With External Link". 
    - In the link at the same location as the one shown in the comp has an external link that links to [www.google.com](http://www.google.com/), and has the button text "Link to Google".
    - In the link after the previous one, add an external link that links to www.google.com, has the override class to not show the external link icon, and has the text, "External Link Override to Not Display"
    - In the link after the previous one, add an internal link that links to www.cancer.gov, has the override class to show the external link icon, and has the text, "Override to Display External Link Icon"

### Acceptance Criteria

```gherkin
Scenario: A Footer link with an external link is viewed on the NCIDS
  Given the external link icon has been added to the NCIDS Documentation Site and NCIDS Storybook for the Footer
    And the Footer link is linked to a url that does not contain `.gov`
  When the Footer link has a linked url that does not contain `.gov`
  Then the Footer link will appear with the external link icon
    And the icon on the Footer link will match the comp (see links above)
    And the external link icon will remain center aligned vertically on the line of text it is on
  When the text of the Footer link wraps
    And the width of the Footer link is only narrow enough where the external link icon should wrap alone
  Then the last word of the Footer link will wrap to the next line with the external link icon
  When a user wants to override the behavior of a Footer link with a link that does not contain `.gov` to not display the external link icon
  Then they will override the behavior with an override class
    And the external link icon will not be displayed
  When a user has a Footer link with a link that contains `.gov`
    And they want to override the behavior to include the external link icon
  Then they will override the behavior with an override class
    And the external link icon will be displayed
```
